dimagi/commcare-cloud
# Enables multiplexer for case search index

Enables multiplexer for case search index so that it can be reindexed.
##### Environments Affected

Production


### Steps to enable and verify multiplexer on Production

This can be rolled out by updating the config and restarting the services
```
cchq --control production update-config
cchq --control production service commcare restart 
```

After the services are restarted, ensure that multiplexer is enabled on the index. You can do this by logging into django shell
```
cchq production django-manage shell
```

Run the following snippet

```
from corehq.apps.es import case_search_adapter
from corehq.apps.es.client import ElasticMultiplexAdapter
type(case_search_adapter)  == ElasticMultiplexAdapter
```

The above snippet should return `True`

You can also run a count query on secondary index to see that documents are getting populated

```
case_saerch_adapter.secondary.count({})
```
